File: java/src/test/java/fun/pierre/lists/P06Test.java
package fun.pierre.lists;

import org.junit.Test;
import static java.util.Arrays.asList;
import static org.hamcrest.CoreMatchers.is;
import static org.junit.Assert.assertThat;

public class P06Test {
  @Test
  public void shoulTestPalindromIntList() {
    assertThat(P06.isPalindrome(asList(1, 2, 3, 2, 1)), is(true));
    assertThat(P06.isPalindrome(asList(1, 2, 3, 4, 5)), is(false));
  }

  @Test
  public void shoulTestPalindromCharList() {
    assertThat(P06.isPalindrome(asList('a', 'b', 'b', 'a')), is(true));
    assertThat(P06.isPalindrome(asList('a', 'b', 'c', 'd')), is(false));
  }
}

File: scala/src/test/scala/fun/pierre/lists/P06Test.scala
package fun.pierre.lists

import org.scalatest.{FunSpec, Matchers}

class P06Test extends FunSpec with Matchers {

  describe("P06 is a palindrome") {
    it("P06 palindrom char element") {
      P06.isPalindrome(List("a", "b", "b", "a")) should be(true)
      P06.isPalindrome(List("a", "b", "b", "d")) should be(false)
    }

    it("P06 palindrom int element") {
      P06.isPalindrome(List(1, 2, 3, 2, 1)) should be(true)
      P06.isPalindrome(List(1, 2, 3, 4, 5)) should be(false)
    }
  }
}
